src/utils/search/index.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * This function employs binary search to search for the index of a line number within the
+ * processed log lines array.
+ * @param processedLines - processed log lines
+ * @param lineNumber - line number to search for
+ * @param start - the point in the array from which to start searching
+ * @param end - the point in the array from which to stop searching
+ * @returns the index of the line number within the array if it was found, otherwise undefined
+ */
+export const findLineIndex = (
+  processedLines: (number | number[])[],
+  lineNumber: number,
+  start: number = 0,
+  end: number = processedLines.length - 1
+): number | undefined => {
+  // Line number was not found in the array.
+  if (start > end) {
+    return undefined;
+  }
+
+  const midIdx = Math.floor((start + end) / 2);
+  const midItem = processedLines[midIdx];
+
+  // If the item is a collapsed row
+  if (Array.isArray(midItem)) {
```

Comment:
I assume you are talking about `L28`: while I agree that it would (or rather should) never happen, I still think it is good to cover the case. Especially if this function could potentially be reusable for something else.